ontopy/excelparser.py

```diff
@@ -0,0 +1,267 @@
+    base_iri_from_metadata: bool = True,
+    catalog: dict = None,
+) -> Tuple[owlready2.Ontology, dict]:
+    """
+    Creates an ontology from an excelfile.
+
+    catalog is dict of imported ontologies with key name and value path
+    """
+    # Read datafile TODO: Some magic to identify the header row
+    conceptdata = pd.read_excel(
+        excelpath, sheet_name=concept_sheet_name, skiprows=[0, 2]
+    )
+    metadata = pd.read_excel(excelpath, sheet_name=metadata_sheet_name)
+    return create_ontology_from_pandas(
+        conceptdata, metadata, base_iri, base_iri_from_metadata, catalog
+    )
+
+
+def create_ontology_from_pandas(  # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+    data: pd.DataFrame,
+    metadata: pd.DataFrame,
+    base_iri: str = "http://emmo.info/emmo/domain/onto#",
+    base_iri_from_metadata: bool = True,
+    catalog: dict = None,
+) -> Tuple[owlready2.Ontology, dict]:
+    """
+    Create an ontology from a pandas DataFrame
+    """
+
+    # Remove Concepts without prefLabel and make all to string
+    data = data[data["prefLabel"].notna()]
+    data = data.astype({"prefLabel": "str"})
+
+    # Make new ontology
+    world = World()
+    onto = world.get_ontology(base_iri)
+
+    onto, catalog = get_metadata_from_dataframe(metadata, onto)
+
+    # base_iri from metadata if it exists and base_iri_from_metadata
+    if not base_iri_from_metadata:
+        onto.base_iri = base_iri
+
+    # have to decide how to add metadata and imports etc.
+    # base_iri to be added from excel (maybe also possibly argument?)
+    # onto = world.get_ontology(base_iri)
+
+    onto.sync_python_names()
+    with onto:
+        # loop through the rows until no more are added
+        new_loop = True
+        final_loop = False
+        while new_loop:
+            number_of_added_classes = 0
+            for _, row in data.iterrows():
+                name = row["prefLabel"]
+                try:
+                    if isinstance(
+                        onto.get_by_label(name), owlready2.ThingClass
+                    ):
+                        continue
+                except NoSuchLabelError:
+                    pass
+
+                parent_names = str(row["subClassOf"]).split(";")
+
+                try:
+                    parents = [onto.get_by_label(pn) for pn in parent_names]
+                except NoSuchLabelError:
+                    if final_loop is True:
+                        parents = onto.EMMO
+
+                        warnings.warn(
+                            "Missing at least one of the defined parents. "
+                            f"Concept: {name}; Defined parents: {parent_names}"
+                        )
+                        new_loop = False
+                    else:
+                        continue
+
+                concept = onto.new_entity(name, parents)
+
+                elucidation = row["Elucidation"]
```

Comment:
If you want to reduce the amount of repeated code, you can define something like
```python
def append_literal_cell(dest, colname, row, split=';'):
    """Appends literal cell in column `colname` of `row` to destination
    list `dest`.  If split is a character, it will be used to split
    the literal before adding them separately to dest.
    """
    values = row[name]
    if isinstance(values, str):
        if split:
            dest.extend([english(v) for v in values])
        else:
            dest.append(english(values))
```
and then do
```python
    append_literal_cell(concept.elucidation, "Elucidation", row, split=None)
    append_literal_cell(concept.example, "Example", row)
    append_literal_cell(concept.comment, "Comment", row)
    ...
```